src/chain.h

```diff
@@ -217,9 +217,19 @@ class CBlockIndex
     std::vector<unsigned char> vStakeModifier{};
     unsigned int nFlags{0};
 
+    //! Change in value held by the Sapling circuit over this block.
+    //! Not a boost::optional because this was added before Sapling activated, so we can
+    //! rely on the invariant that every block before this was added had nSaplingValue = 0.
+    CAmount nSaplingValue{0};
+
+    //! (memory only) Total value held by the Sapling circuit up to and including this block.
+    //! Will be boost::none if nChainTx is zero.
+    boost::optional<CAmount> nChainSaplingValue{boost::none};
```

Comment:
If you include our `optional.h` header, can use our wrapper template and substitute
```cpp
Optional<CAmount> nChainSaplingValue{nullopt};
```